include/Options/Business/Nav_Menus.php

```diff
@@ -73,6 +83,78 @@ protected function get_data_structure(): array {
 		);
 	}
 
+	/**
+	 * Sanitizes option's value.
+	 * Can populate the `$errors` property with blocking and non-blocking errors: in case of non-blocking errors,
+	 * the value is sanitized and can be stored.
+	 *
+	 * @since 3.7
+	 *
+	 * @param array   $value   Value to sanitize.
+	 * @param Options $options All options.
+	 * @return array|WP_Error The sanitized value. An instance of `WP_Error` in case of blocking error.
+	 *
+	 * @phpstan-return NavMenusValue|WP_Error
+	 */
+	protected function sanitize( $value, Options $options ) {
+		global $polylang;
+
+		if ( empty( $polylang ) || ! $polylang->model->are_languages_ready() ) {
+			// Access to global `$polylang` is required.
+			_doing_it_wrong(
+				__METHOD__,
+				esc_html( sprintf( 'The option \'%s\' cannot be set before the hook \'pll_init\'.', static::key() ) ),
+				'3.7'
+			);
+			/** @phpstan-var NavMenusValue */
+			return $this->get();
+		}
```

Comment:
I don't see where `$polylang` is used outside this block. To avoid using a global, we could change `Languages::set_ready()` to fire an action `pll_languages_ready` instead of setting a boolean flag. The check could then be made with `did_action()`. Also this check could be moved to `Abstract_Option` to avoid duplicate code.

Reply:
Check factorized in [fe2f891](https://github.com/polylang/polylang/pull/1610/commits/fe2f891cf27fda8d354237f4d221493e3259a9e6).

Reply:
I removed these checks about languages being ready in 85ecc97.
They are not needed anymore because we retrieve the languages list directly from the language terms now.

Note: I tried your suggestion and it breaks some tests, probably because we use more than one model (so the 1st model "readiness" breaks the 2nd model's).